src/eradiate_plugins/media/piecewise.cpp

```diff
@@ -0,0 +1,512 @@
+#include <mitsuba/core/frame.h>
+#include <mitsuba/core/properties.h>
+#include <mitsuba/core/spectrum.h>
+#include <mitsuba/core/warp.h>
+#include <mitsuba/render/interaction.h>
+#include <mitsuba/render/medium.h>
+#include <mitsuba/render/phase.h>
+#include <mitsuba/render/sampler.h>
+#include <mitsuba/render/scene.h>
+#include <mitsuba/render/volume.h>
+#include <mitsuba/core/distr_1d.h>
+
+NAMESPACE_BEGIN(mitsuba)
+
+/**!
+
+.. _medium-piecewise:
+
+Piecewise medium (:monosp:`piecewise`)
+-----------------------------------------------
+
+.. pluginparameters::
+
+ * - albedo
+   - |float|, |spectrum| or |volume|
+   - Single-scattering albedo of the medium (Default: 0.75).
+   - |exposed|, |differentiable|
+
+ * - sigma_t
+   - |float|, |spectrum| or |volume|
+   - Extinction coefficient in inverse scene units (Default: 1).
+     The supplied grid must be of shape [1,1,n].
+   - |exposed|, |differentiable|
+
+ * - scale
+   - |float|
+   - Optional scale factor that will be applied to the extinction parameter.
+     It is provided for convenience when accommodating data based on different
+     units, or to simply tweak the density of the medium. (Default: 1)
+   - |exposed|
+
+ * - sample_emitters
+   - |bool|
+   - Flag to specify whether shadow rays should be cast from inside the volume (Default: |true|)
+     If the medium is enclosed in a :ref:`dielectric <bsdf-dielectric>` boundary,
+     shadow rays are ineffective and turning them off will significantly reduce
+     render time. This can reduce render time up to 50% when rendering objects
+     with subsurface scattering.
+
+ * - (Nested plugin)
+   - |phase|
+   - A nested phase function that describes the directional scattering properties of
+     the medium. When none is specified, the renderer will automatically use an instance of
+     isotropic.
+   - |exposed|, |differentiable|
+
+
+This plugin provides a 1D heterogenous medium implementation (plane parallel geometry),  
+which acquires its data from nested volume instances. These can be constant, 
+use a procedural function, or fetch data from disk, e.g. using a 3D grid, as long as the
+underlying grid has the shape [1,1,N], with N the number of layers on the z axis.
```

Comment:
I assume that there is an assumption on what the vertical axis is? If so, it's good to mention it in the documentation.